source/__tests__/integration/new-profile.test.js

```diff
@@ -41,19 +41,32 @@ describe('Adding and removing a new Profile', () => {
         await expect(popover).toBe('Link to more information about profile')
       })
 
-      it('validates with checkURL', async () => {
+      it('with function checkURL', async () => {
         const ng_blur = await page.$eval(source_input_sel, e => e.getAttribute('ng-blur'))
         await expect(ng_blur).toBe('checkURL()')
       })
+
+      it('displays error if source and user switches focus', async () => {
+        await page
+          .type(source_input_sel, '')
+          .catch(error => console.log(`promise error for : ${error}`))
+        await page.keyboard.press(`Tab`)
+        await expect_value_in_selector_textContent('#alertBox > div > div > div.modal-header > h3', 'Error!')
+        const invalid_url_class = page.$('input#resourceURI', e => e.getAttribute('ng-invalid-url'))
+        expect(invalid_url_class).toBeDefined()
+        await page.$eval("button#alertClose", e => e.click())
+        await page.waitFor(1000)
```

Comment:
another unconditional waitFor() and it's the last thing in a test ... does there need to be a conditional waitFor() at the beginning of the next test?  or is this unnecessary?  Did with line 57 - clicking a button at the end, but not checking anything, is an odd way to end a test ... what am I missing?

Reply:
So the close button and waitFor were necessary to reset the UI for other tests in this suite. Otherwise any tests that occurred afterwards was failing because alertBox was still active. I'll change the scope for this describe block so it isn't trampling on other tests.